tests/lto_and_gold.c

```diff
@@ -0,0 +1,20 @@
+// Test LTO support within ld.gold
+// REQUIRES: clang, lld
```

Comment:
This test does not actually uses lld, right? If so, let's remove it so the test runs in platforms that don't support lld (such as s390x)